src/test/bridges/yearn/YearnBridgeE2E.t.sol

```diff
@@ -0,0 +1,223 @@
+import {IYearnRegistry} from "../../../interfaces/yearn/IYearnRegistry.sol";
+import {YearnBridge} from "../../../bridges/yearn/YearnBridge.sol";
+import {ErrorLib} from "../../../bridges/base/ErrorLib.sol";
+
+contract YearnBridgeE2ETest is BridgeTestBase {
+    using SafeERC20 for IERC20;
+
+    struct Balances {
+        uint256 underlyingBefore;
+        uint256 underlyingMid;
+        uint256 underlyingEnd;
+        uint256 sharesBefore;
+        uint256 sharesMid;
+        uint256 sharesEnd;
+    }
+
+    IERC20 public constant YVETH = IERC20(0xa258C4606Ca8206D8aA700cE2143D7db854D168c);
+    IERC20 public constant OLD_YVETH = IERC20(0xa9fE4601811213c340e850ea305481afF02f5b28);
+    IERC20 public constant WETH = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    IERC20 public constant DAI = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    uint256 public constant SAFE_MAX = 79228162514264337593543950334;
+
+    YearnBridge internal bridge;
+    uint256 internal id;
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+
+    function setUp() public {
+        bridge = new YearnBridge(address(ROLLUP_PROCESSOR));
+        bridge.preApproveAll();
+        vm.label(address(bridge), "YEARN_BRIDGE");
+        vm.deal(address(bridge), 0);
+
+        vm.prank(MULTI_SIG);
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 5000000);
+
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+    }
+
+    function testERC20DepositAndWithdrawal(uint256 _depositAmount, uint256 _withdrawAmount) public {
+        IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+        address vault = _registry.latestVault(address(DAI));
+        uint256 availableDepositLimit = IYearnVault(vault).availableDepositLimit();
+        if (availableDepositLimit > 0) {
+            _depositAmount = bound(_depositAmount, 1e17, availableDepositLimit);
+            _withdrawAmount = bound(_withdrawAmount, 1e17, _depositAmount);
+            _depositAndWithdrawERC20(vault, _depositAmount, _withdrawAmount);
+        }
+    }
+
+    // function testMoreERC20DepositAndWithdrawal() public {
+    //     IYearnRegistry _registry = bridge.YEARN_REGISTRY();
+    //     uint256 numTokens = _registry.numTokens();
+    //     for (uint256 i; i < numTokens;) {
+    //         address token = _registry.tokens(i);
+    //         if (
+    //             token == address(WETH) ||
+    //             token == 0xC011a73ee8576Fb46F5E1c5751cA3B9Fe0af2a6F //SNX, packed slot not supported
+    //         ) {
+    //             unchecked {
+    //                 ++i;
+    //             }
+    //             continue;
+    //         }
+    //         address vault = _registry.latestVault(token);
+    //         uint256 availableDepositLimit = IYearnVault(vault).availableDepositLimit();
+    //         if (availableDepositLimit > 0) {
+    //             uint256 min = 1e6;
+    //             if (availableDepositLimit < min) {
```

Comment:
Unclear what this if is for, would it not be easier to just define `1e2` as min in `bound`?